src/gui/properties/speedplotview.cpp

```diff
@@ -40,26 +40,19 @@
 
 namespace
 {
-    enum PeriodInSeconds
-    {
-        MIN1_SEC = 60,
-        MIN5_SEC = 5 * 60,
-        MIN30_SEC = 30 * 60,
-        HOUR6_SEC = 6 * 60 * 60,
-        HOUR12_SEC = 12 * 60 * 60,
-        HOUR24_SEC = 24 * 60 * 60
-    };
-
-    const int MIN5_BUF_SIZE = 5 * 60;
-    const int MIN30_BUF_SIZE = 5 * 60;
-    const int HOUR6_BUF_SIZE = 5 * 60;
-    const int HOUR12_BUF_SIZE = 10 * 60;
-    const int HOUR24_BUF_SIZE = 10 * 60;
-    const int DIVIDER_30MIN = MIN30_SEC / MIN30_BUF_SIZE;
-    const int DIVIDER_6HOUR = HOUR6_SEC / HOUR6_BUF_SIZE;
-    const int DIVIDER_12HOUR = HOUR12_SEC / HOUR12_BUF_SIZE;
-    const int DIVIDER_24HOUR = HOUR24_SEC / HOUR24_BUF_SIZE;
-
+    using namespace std::chrono_literals;
+
+    const SpeedPlotView::Milliseconds DURATION_MIN1 {1min};
+    const SpeedPlotView::Milliseconds DURATION_MIN5 {5min};
+    const SpeedPlotView::Milliseconds DURATION_MIN30 {30min};
+    const SpeedPlotView::Milliseconds DURATION_HOUR6 {6h};
+    const SpeedPlotView::Milliseconds DURATION_HOUR12 {12h};
+    const SpeedPlotView::Milliseconds DURATION_HOUR24 {24h};
+    const SpeedPlotView::Milliseconds RESOLUTION_MIN5 {1s};
+    const SpeedPlotView::Milliseconds RESOLUTION_MIN30 {6s};
+    const SpeedPlotView::Milliseconds RESOLUTION_HOUR6 {36s};
+    const SpeedPlotView::Milliseconds RESOLUTION_HOUR12 {72s};
+    const SpeedPlotView::Milliseconds RESOLUTION_HOUR24 {144s};
 
```

Comment:
IMHO directly using numbers seems a better idea, for example, previously 6-hour and 12-hour graphs were using the same resolution i.e 72 (DIVIDER_6HOUR == DIVIDER_12HOUR). using time points solves such bugs.